packages/hoppscotch-cli/src/utils/request.ts

```diff
@@ -358,3 +369,21 @@ export const getRequestMetrics = (
       hasReqErrors ? { failed: 1, passed: 0 } : { failed: 0, passed: 1 },
     (requests) => <RequestMetrics>{ requests, duration }
   );
+
+/**
+ * Custom axios wrapper to support extra request configurations such as
+ * delay between consecutive requests.
+ * @param config Current request configuration with additional parameters.
+ * @returns Manipulated axios promise based on additional parameters.
+ */
+const axiosDelay = (config: RequestConfig): AxiosPromise => {
+  return new Promise(async (resolve, reject) => {
+    try {
+      setTimeout(() => {
+        resolve(axios(config));
+      }, config.delay);
+    } catch (error) {
+      reject(error);
+    }
+  });
```

Comment:
The try-catch is redundant and will not catch anything. as the setTimeout will run and the block execution will be complete before the axios execution occurs. 

Along with that the unused async/await on the async function.

Also, I think its a bad idea to delay axios calls, that feels weird, let the delay apply to one hoppscotch request instead of an axios one.